optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/uni/DroolsScoringUniConstraintStream.java

```diff
@@ -98,8 +98,9 @@ private DroolsScoringUniConstraintStream(DroolsConstraintFactory<Solution_> cons
 
     @Override
     public void createRuleItemBuilders(List<RuleItemBuilder<?>> ruleItemBuilderList,
-            Global<? extends AbstractScoreHolder> scoreHolderGlobal,
-            Declaration<A> aVar, PatternDSL.PatternDef<A> parentPattern) {
+            Global<? extends AbstractScoreHolder> scoreHolderGlobal) {
+        PatternDSL.PatternDef<A> parentPattern = getPattern();
+        Declaration<A> aVar = getVariableDeclaration();
```

Comment:
Ah, this refactor is a problem :)
I intentionally passed the variable declarations - instead of fetching them like this - despite it causing less readable code. The motivation:

from(A.class) // Uni<A>
.groupBy(A::getB) // Uni<B>
.penalize(); // The variable declaration should be only B. The one for A got burried.

from(A.class) // Uni<A>
.join(B.class) // Bi<A, B>
.groupBy((A a, B b) -> a.getCMap().get(b), sum(a.price * b.vat)) // Bi<C, int>
.penalize // The variable declaration should be only C I think?

Reply:
@ge0ffrey This is what I mean when I say we have several fundamentally different `groupBy` methods. I don't see a way around that, other than implementing them with different `ConstraintStream` implementation classes. I'll cross that bridge when I get there, OK?

Reply:
I think you 'll have to undo this method refactor, the one done in this case, but I am ok with leaving it like this until then.